solution/manual-offset/src/main/java/clients/ManualOffsetManagement.java

```diff
@@ -0,0 +1,55 @@
+package clients;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+
+public class ManualOffsetManagement {
+    public static void main(String[] args) throws IOException {
+        final String OFFSET_FILE_PREFIX = "offset_";
```

Comment:
Minor - could possibly make a slight modification here to point to a mounted host volume to a path in the container. For example:

```java
final String OFFSET_FILE_PREFIX = "/offsets/offset_";
```

Then, when running the app in Docker:

```docker
docker container run --rm \
--net solution_confluent \
-v "$PWD"/build/libs/app.jar:/app.jar -v "$PWD"/offsets:/offsets \
openjdk:8-jre-alpine java -jar /app.jar
```